synapse/storage/databases/main/events_worker.py

```diff
@@ -729,12 +731,16 @@ async def get_missing_events_from_db() -> Dict[str, EventCacheEntry]:
 
         return event_entry_map
 
-    def _invalidate_get_event_cache(self, event_id: str) -> None:
-        self._get_event_cache.invalidate((event_id,))
+    async def _invalidate_get_event_cache(self, event_id: str) -> None:
+        await self._get_event_cache.invalidate((event_id,))
+        self._invalidate_local_get_event_cache(event_id)
```

Comment:
This means that we end up with two `.invalidate` calls on the underlying `LruCache`. It might be better just to:

```suggestion
        self._event_ref.pop(event_id, None)
        self._current_event_fetches.pop(event_id, None)
```

Reply:
https://github.com/matrix-org/synapse/pull/13242/commits/ec33b99b12619016b2063db4508e7af15185c1d5